### PR TITLE
Support specifying licenses for Google Compute images

### PR DESCRIPTION
**Pull request by seanmalloy**
This is needed to enable features such as the nested virtualization:
https://cloud.google.com/compute/docs/instances/enable-nested-virtualization-vm-instances

### CODE REVIEW

**Comment by seanmalloy** [comment]
This pull request supersedes  #5716. All of the code is the same. I just added the required documentation for the new `image_licenses` option.

Thanks to @petrhosek for writing all of the code!

**Comment by seanmalloy** [comment]
Here is the packer configuration file I used to test with. You will need to replace the value for `project_id` with your GCP project  ID.

```
{
  "builders": [
    {
      "type": "googlecompute",
      "account_file": "account.json",
      "project_id": "smalloy-sbx1",
      "zone": "us-central1-a",
      "source_image_family": "centos-7",
      "ssh_username": "packer",
      "image_licenses": ["projects/vm-options/global/licenses/enable-vmx"]
    }
  ]
}
```

**Comment by seanmalloy** [comment]
All of the tests pass  on my machine when running `make test` using Go 1.9.2 on Linux, so I'm not sure why the Travis CI tests are failing.

File: website/source/docs/builders/googlecompute.html.md [hunk 1]
```diff
@@ -122,25 +125,50 @@ user used to connect in a startup-script.
...
+      "account_file": "account.json",
+      "project_id": "my project",
+      "source_image_family": "centos-7",
+      "ssh_username": "packer",
+      "zone": "us-central1-a",
+      "image_licenses": ["projects/vm-options/global/licenses/enable-vmx"]
+    }
+  ]
+}
+
```
Comment (rickard-von-essen) [inline website/source/docs/builders/googlecompute.html.md]: Nit pick, extra NL

Comment (seanmalloy) [reply website/source/docs/builders/googlecompute.html.md]: Fixed

**Comment by mwhooker** [comment]
thanks!

**Pull request merged**